packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2639,13 +2642,58 @@ class InputDecoration {
   /// See [Icon], [ImageIcon].
   final Widget? icon;
 
-  /// Text that describes the input field.
+  /// Optional widget that describes the input field.
+  ///
+  /// This can be used, for example, to add multiple [TextStyle]'s to a label that would
+  /// otherwise be specified using [labelText], which only takes one [TextStyle].
```

Comment:
Nit: Do you think it would be useful to mention that `label` will move up when the field is focused?  You could use a macro to get the second paragraph of labelText's docs.

Reply:
Yeah this would be good to mention i'll add it.